src/context/auth.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useContext, useReducer, useCallback } from "react";
+import axios from "axios";
+import { getUiUrl } from "utils/getEnvironmentVariables";
+
+interface State {
+  isAuthenticated: boolean;
+}
+
+const defaultState: State = {
+  isAuthenticated: false
+};
+
+type Action = { type: "authenticate" } | { type: "deauthenticate" };
```

Comment:
"authenticate" and "deauthenticate" should be consts

Reply:
these are not strings in JS, they are actually types. therefore they will autocomplete as though they are consts. try it out

Reply:
Shouldn't this be an enum instead though?